docs/modules/angular/pages/cookbooks/vanilla-to-onecx/migrate-to-onecx.adoc

```diff
@@ -0,0 +1,279 @@
+In order to expose this file and ensure correct dependency sharing between the app and OneCX, a Webpack config file containing specific configuration properties has to be created.
+
+The file, `webpack.config.js`, which should be placed in the root folder of the application must contain the following configuration:
+
+[source,typescript]
+const { ModifyEntryPlugin } = require('@angular-architects/module-federation/src/utils/modify-entry-plugin')
+const { share, withModuleFederationPlugin } = require('@angular-architects/module-federation/webpack')
+const config = withModuleFederationPlugin({
+  name: 'YOUR_MFE_NAME',
+  filename: 'remoteEntry.js',
+  exposes: {
+    './RemoteModule': 'src/remote-main.ts',
+  },
+  shared: share({
+    '@angular/core': { requiredVersion: 'auto', includeSecondaries: true },
+    '@angular/common': { requiredVersion: 'auto', includeSecondaries: { skip: ['@angular/common/http/testing'] } },
+    '@angular/common/http': { requiredVersion: 'auto', includeSecondaries: true },
+    '@angular/forms': { requiredVersion: 'auto', includeSecondaries: true },
+    '@angular/platform-browser': { requiredVersion: 'auto', includeSecondaries: true },
+    '@angular/router': { requiredVersion: 'auto', includeSecondaries: true },
+    rxjs: { requiredVersion: 'auto', includeSecondaries: true },
+    '@onecx/angular-integration-interface': { requiredVersion: 'auto', includeSecondaries: true },
+    '@onecx/angular-webcomponents': { requiredVersion: 'auto', includeSecondaries: true },
+  }),
+})
+config.devServer = { allowedHosts: 'all' }
+const plugins = config.plugins.filter((plugin) => !(plugin instanceof ModifyEntryPlugin))
+module.exports = {
+  ...config,
+  plugins,
+  output: { uniqueName: 'YOUR_MFE_NAME', publicPath: 'auto' },
+  experiments: { ...config.experiments, topLevelAwait: true },
+  optimization: { runtimeChunk: false, splitChunks: false }
+}
+
+NOTE: This file is meant to be used as a minimal starting point. Depending on the application the contents of the file, especially the shared config, may need to be adapted.
+
+To allow for a custom Webpack configuration in production environments it's a good practice to additionally create a `webpack.prod.js` file that will extend the base config:
+[source, typescript]
+module.exports = require('./webpack.config')
+
+If necessary, this file can be used to overwrite or extend the contents of the base config with production specific values.
+
+=== Add environment file `env.json`
+During the loading of an app, OneCX will try to load it's environment file. To avoid any errors, a file called `env.json` should be accessible at http://localhost:4200/remoteEntry. This can be achieved by adding the respective file to the assets or public folder of the application. The file can be used to pass environment variables to OneCX. For now it can just contain an empty object ({}).
```

Comment:
Move content to code block. Otherwise, someone could copy the ( ) along with the curlies?